src/server/session.ts

```diff
@@ -870,20 +868,37 @@ namespace ts.server {
         }
 
         /** It is the caller's responsibility to verify that `!this.suppressDiagnosticEvents`. */
-        private updateErrorCheck(next: NextStep, checkList: PendingErrorCheck[], ms: number, requireOpen = true) {
+        private updateErrorCheck(next: NextStep, checkList: readonly string[] | readonly PendingErrorCheck[], ms: number, requireOpen = true) {
             Debug.assert(!this.suppressDiagnosticEvents); // Caller's responsibility
 
             const seq = this.changeSeq;
             const followMs = Math.min(ms, 200);
 
             let index = 0;
+            const goNext = () => {
+                index++;

```

Comment:
seems like there are places where index doesn't get incremented now that it's inside `goNext`. For example, when `!project.containsFile(fileName, requireOpen)`. Is this important?

Reply:
Since only goToNext is the only one that reschedules the next work item.. Thats the only one it needs to increment index.. Because in rest of the cases the we dont go to next work item at all.